### PR TITLE
Dashboard component: Add rendering worker and support for export mode

### PR DESCRIPTION
**Pull request by kandl**
- Implement rendering commands/events/hook/worker, so dashboard is able to notify that its rendered
- Add support for background workers
- Hide relevant components in export mode, apply required styles
- Tests
- Minor bug fixing/renaming

JIRA: RAIL-3554

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4574/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6195/ : SUCCESS in 16m 45s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4577/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6198/ : SUCCESS in 16m 50s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4578/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6199/ : SUCCESS in 14m 09s

File: libs/sdk-ui-dashboard/src/model/commandHandlers/render/renderingWorker.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    );
+
+/**
+ * @internal
+ */
+export interface RenderingWorkerConfiguration {
+    /**
+     * Maximum time limit for rendering the dashboard.
+     *
+     * Default: 60000 (6s).
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/render/renderingWorker.ts:31]: This comment is probably wrong either 60 seconds or 6_000 as the default.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/render/tests/__snapshots__/renderingWorker.test.ts.snap [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+exports[`renderingWorker async rendering resolution should accept new async rendering request, when it's fired within time limit after resolution 1`] = `
+Array [
+  Object {
+    "correlationId": undefined,
+    "type": "GDC.DASH/EVT.RENDER.REQUESTED",
+  },
+  Object {
+    "correlationId": undefined,
+    "type": "GDC.DASH/EVT.FILTER_CONTEXT.DATE_FILTER.VALIDATION.FAILED",
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/render/tests/__snapshots__/renderingWorker.test.ts.snap:11]: This is suspicious a bit, why does the validation fail?

File: libs/sdk-ui-dashboard/src/model/commandHandlers/render/renderingWorker.ts (hunk 1 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/render/renderingWorker.ts:31]: Good catch, I'll correct it.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/render/tests/__snapshots__/renderingWorker.test.ts.snap (hunk 2 above)
Comment (kandl) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/render/tests/__snapshots__/renderingWorker.test.ts.snap:11]: Good point, I'll take a look.

File: libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx [hunk 3]
```diff
@@ -28,6 +29,11 @@ export const DefaultFilterBarInner = (): JSX.Element => {
     const availableGranularities = useDashboardSelector(selectEffectiveDateFilterAvailableGranularities);
     const dateFilterOptions = useDashboardSelector(selectEffectiveDateFilterOptions);
     const dateFilterMode = useDashboardSelector(selectEffectiveDateFilterMode);
+    const isExport = useDashboardSelector(selectIsExport);
+
+    if (isExport) {
+        return <React.Fragment />;
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/DefaultFilterBar.tsx:35]: Detail: we should probably use HiddenFilterBar for this.

File: libs/sdk-ui-dashboard/src/presentation/topBar/topBar/DefaultTopBar.tsx [hunk 4]
```diff
@@ -35,6 +36,12 @@ const TopBarCore = (): JSX.Element => {
  * @internal
  */
 export const DefaultTopBarInner = (): JSX.Element => {
+    const isExport = useDashboardSelector(selectIsExport);
+
+    if (isExport) {
+        return <React.Fragment />;
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/topBar/topBar/DefaultTopBar.tsx:42]: Detail: please use HiddenTopBar for this

**Review by no23reason — APPROVED** [review]
LGTM

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1569

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1300/ : FAILURE in 2m 38s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1576/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/350/ : SUCCESS in 11m 14s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6202/ : SUCCESS in 17m 52s

**Pull request merged**